Engine/ac/dialog.cpp

```diff
@@ -879,8 +879,8 @@ bool DialogOptions::Run()
         run_function_on_non_blocking_thread(&runDialogOptionRepExecFunc);
       }
 
-      if (kbhit()) {
-        int gkey = getch();
+      int gkey;
+      if (run_service_key_controls(gkey)) {

```

Comment:
Hmm, could they be logically split in similar way?
Basically I am forcing engine to query keys using same shared function everytime.

Reply:
I only made a quick look, and it seems to me the change may be trivial:

You could pick out following part:
<pre>
    SDL_Event kgn = getTextEventFromQueue();
    if (kgn.type == 0) { return; }
    
    if (isScancode(kgn, SDL_SCANCODE_LCTRL) || isScancode(kgn, SDL_SCANCODE_RCTRL) || isScancode(kgn, SDL_SCANCODE_LALT) || isScancode(kgn, SDL_SCANCODE_RALT) || isScancode(kgn, SDL_SCANCODE_MODE)) {
        SDL_Keymod mod_state = SDL_GetModState();
        if ( (mod_state & KMOD_CTRL) && (mod_state & (KMOD_ALT|KMOD_MODE)) ) {
            toggle_mouse_lock();
            return;
        }
    }

    // LAlt or RAlt + Enter
    if (kgn.type == SDL_KEYDOWN && (kgn.key.keysym.mod & KMOD_ALT) && (kgn.key.keysym.scancode == SDL_SCANCODE_RETURN))
    {
        engine_try_switch_windowed_gfxmode();
        return;
    }
</pre>
into separate function called run_service_key_controls and call it every time instead of just SDL_Event gkey = getTextEventFromQueue();